build/package/send-slack-notification.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+# Copyright 2021 MongoDB Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -Eeou pipefail
```

Comment:
[nit] similar to above, let's make this silent so it doesn't log keys

Reply:
[q] wh aren't we setting `silent` also for the other evergreen task that uses keys? 

- https://github.com/mongodb/mongocli/blob/a2e58587923a453e6e0df435835437d5f2c0fde2/build/ci/evergreen.yml#L88
- https://github.com/mongodb/mongocli/blob/a2e58587923a453e6e0df435835437d5f2c0fde2/build/ci/evergreen.yml#L142
- https://github.com/mongodb/mongocli/blob/a2e58587923a453e6e0df435835437d5f2c0fde2/build/ci/evergreen.yml#L417

Reply:
because those task don't use the keys as part of their action, the keys are part of the env variable so they won't leak to logs, this task directly uses them like `curl --header "Api-User:${evergreen_user:?}" \` which combined with `set -Eeou` will log that line and hence log them into evergreen logs

Reply:
Additionally check how we do `set -Eeou pipefail` after exposing the envs precisely to avoid that issue https://github.com/mongodb/mongocli/blob/bdf871339c62c69bf580030332958ce5b120c708/build/package/curator-push.sh#L32